### ISSUE TITLE
Error : Cannot read properties of undefined (reading 'documents')

### ISSUE THREAD

**Issue by JoeFieck** [Azure/azure-cosmosdb-ads-extension#112]
Hello.  I had opened https://github.com/microsoft/azuredatastudio/issues/25522 and was instructed to open here for support.

Upgrade ADS to 1.48.0
Install the Azure Cosmos DB extension.
Make a connection to Azure Cosmos DB using a connection string.
Notice that the UI correctly shows the same list of containers showing from Data Explore from withing Azure directly.

Open a query and run:
select * from c

You may see a message saying:
**Cannot read properties of undefined (reading 'documents')**

No data is returned.

What is interesting is that on the bug fix list for 1.48 it shows having fixed an issue with :
Connections Fixed Error: “Cannot read properties of undefined (reading 'serverInfo')”

This message here is the same except for it shows 'serverInfo' where mine shows 'documents'

----------------Here are the system info and package info you may want---------------
Item	Value
CPUs	11th Gen Intel(R) Core(TM) i5-1135G7 @ 2.40GHz (8 x 2419)
GPU Status	2d_canvas: enabled
canvas_oop_rasterization: enabled_on
direct_rendering_display_compositor: disabled_off_ok
gpu_compositing: enabled
multiple_raster_threads: enabled_on
opengl: enabled_on
rasterization: enabled
raw_draw: disabled_off_ok
video_decode: enabled
video_encode: enabled
vulkan: disabled_off
webgl: enabled
webgl2: enabled
webgpu: enabled
Load (avg)	undefined
Memory (System)	15.73GB (6.34GB free)
Process Argv	
Screen Reader	no
VM	0%
Extensions (9)
Extension	Author (truncated)	Version
admin-pack	Mic	0.0.2
admin-tool-ext-win	Mic	0.1.3
agent	Mic	0.49.0
azure-cosmosdb-ads-extension	Mic	0.4.2
azuredatastudio-postgresql	Mic	0.6.0
dacpac	Mic	1.14.0
import	Mic	1.6.1
net-6-runtime	Mic	1.1.0
profiler	Mic	0.14.0

**Comment by JoeFieck** [comment]
![image](https://github.com/Azure/azure-cosmosdb-ads-extension/assets/148320389/d423f4b3-9258-4c8b-a946-60fabd4a553c)

**Comment by JoeFieck** [comment]
![image](https://github.com/Azure/azure-cosmosdb-ads-extension/assets/148320389/97a326b4-fbc5-4f2c-adbe-c6f04aceda95)

**Comment by JoeFieck** [comment]
Hello all.   I just upgrade ADS to 1.48.0 and installed the azure-cosmosdb-ads-extention at 0.4.2.
When I try to run a query I get this "**Cannot read properties of undefined (reading '_documents_')**" error.  
When I launch ADS and connect to my Cosmos db on Azure it will load up the databases and the containers but just will not let me run the basic select statement.

I have seen many others out there seeing the same but everyone's solution is different and usually ends with this going away on its own or needing a reinstallation of the extension.  I have tried that but no luck so far.

What is interesting is there was a bug fixed on ADS Cosmo DB ads ext that fixed an error with:
Error: “**Cannot read properties of undefined (reading '_serverInfo_')”**  .  So only diff is it was "serverInfo" where mine is "documents".  I'm sure they are related.  Please help.   I really need one place to work and ADS is where all my other SQL connections are made.

**Comment by JoeFieck** [comment]
Hi @cheenamalhotra .  I have opened this as you requested...

**Comment by JoeFieck** [comment]
Hi @cheenamalhotra , can we have this assigned and looked at please...

**Comment by cheenamalhotra** [comment]
I don't work on the extension team, tagging @languy to help triage and investigate the issue.

**Comment by JoeFieck** [comment]
Thanks Cheena.

**Comment by JoeFieck** [comment]
Hello @languy .  Had anyone had a minute to look into fixing this on the next release of Azure Data Studio.   Again there was a very similar message fixed on .48 that I think is the same here..

**Comment by JoeFieck** [comment]
@languy and @cheenamalhotra can someone acknowledge any of these is even been seen.  Its so frustrating trying to get something addressed over the internet these days.  You just feel like things go into the black hole and never reach a humans ears.  :-|   

Again I think this is  regression to a bug fixed on version 49.   Can someone look into this or at least tell me we're going to so I don't keep coming back here trying to reach anyone that might have access to investigate...   I would really appreciate it.

**Comment by JoeFieck** [comment]
@languy and @cheenamalhotra can someone acknowledge any of this is even being seen.  Its so frustrating trying to get something addressed over the internet these days.  You just feel like things go into the black hole and never reach a humans ears.  :-|   

Again I think this is  regression to a bug fixed on version 49.   Can someone look into this or at least tell me we're going to so I don't keep coming back here trying to reach anyone that might have access to investigate...   I would really appreciate it.

**Comment by JoeFieck** [comment]
Hello there @languy and @cheenamalhotra .  Just following up on this one again.  Any update?

**Comment by JoeFieck** [comment]
Hi Cheena.  I have opened the new case with the cosmos connector team.

Description and number is:
Error : Cannot read properties of undefined (reading 'documents') #112

I have been updating for status tagging you and languy but have not heard
anything or seen any updates.  Checking again...

**Comment by JoeFieck** [comment]
Hi @languy just following up on this again...

**Comment by languy** [comment]
Hi @JoeFieck, apologies for the delayed response (I was out of office).

The bug is definitely in this extension, not in the ADS 1.48 code. I see some places where this error could happen, let's narrow it down:
* Could you expand the error and report what the details are? (there is an arrow up button ^ next to the close button x in the error dialog box that you reported)
* There is also a developer console that can provide more information:
  * Hit Ctrl-Shift-P to open the ADS command prompt and type: toggle developer tools, select "Developer: Toggle Developer Tools". This will open a new window
  * At the top of that developer console, click on "Console".
  * Now repro the error and copy paste what is logged in the console (namely, we're looking for the same error message in red with the stack trace)
* If you run that query in the portal Data Explorer, do you see any result? Or empty results?

**Comment by JoeFieck** [comment]
Hello @languy .  The steps I follow are to open ADS then right click the db and select "Open Query".
Then I run the "select * from c" query.  The first time I hit run nothing happens but if I hit run a 2nd time I see the dialog below which I have expanded...

![image](https://github.com/Azure/azure-cosmosdb-ads-extension/assets/148320389/aa9ce0d6-8ed8-4b36-9a12-1ffb4877d2aa)

**Comment by JoeFieck** [comment]
Here is the console output you wanted reproducing the error:

<html>
<body>
<div tabindex="-1" class="console-message-wrapper console-from-api console-info-level" style="box-sizing: border-box; min-width: 0px; min-height: 0px; display: flex; border-top-width: ; border-top-style: ; border-top-color: transparent; border-bottom: 1px solid transparent; --console-color-black: #000; --console-color-red: #a00; --console-color-green: #0a0; --console-color-yellow: #a50; --console-color-blue: #00a; --console-color-magenta: #a0a; --console-color-cyan: #0aa; --console-color-gray: #aaa; --console-color-darkgray: #555; --console-color-lightred: #f55; --console-color-lightgreen: #5f5; --console-color-lightyellow: #ff5; --console-color-lightblue: #55f; --console-color-ightmagenta: #f5f; --console-color-lightcyan: #5ff; --console-color-white: #fff; color: rgb(32, 33, 36); font-family: consolas, &quot;lucida console&quot;, &quot;courier new&quot;, monospace; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="console-message" style="box-sizing: border-box; min-width: 0px; min-height: 17px; clear: right; position: relative; padding: 3px 22px 1px 0px; margin-left: 24px; flex: 1 1 auto; display: flex;"><span class="source-code" style="box-sizing: border-box; min-width: 0px; min-height: 0px; font-family: var(--source-code-font-family); white-space: pre-wrap; font-size: var(--source-code-font-size) !important; flex: 1 1 auto; line-height: 1.2;"><span class="console-message-text" style="box-sizing: border-box; min-width: 0px; min-height: 0px;">onSubmitQuery <div class="console-view-object-properties-section object-value-object source-code" style="flex: 1 1 auto; padding: 0px; --monospace-font-size: 12px; --monospace-font-family: consolas,lucida console,courier new,monospace; --source-code-font-size: 12px; --source-code-font-family: consolas,lucida console,courier new,monospace; box-sizing: border-box; min-width: 0px; min-height: 0px; font-family: var(--source-code-font-family); white-space: pre-wrap; font-size: var(--source-code-font-size) !important; position: relative; vertical-align: baseline; color: inherit; display: inline-block; overflow-wrap: break-word; max-width: 100%; line-height: 1.2;"><div class="tree-outline-disclosure tree-outline-disclosure-hide-overflow" style="box-sizing: border-box; min-width: 0px; min-height: 0px;"><ol class="tree-outline hide-selection-when-blurred source-code object-properties-section" role="tree" tabindex="-1" style="box-sizing: border-box; min-width: 0px; min-height: 0px; font-family: var(--source-code-font-family); white-space: pre-wrap; font-size: var(--source-code-font-size) !important; padding: 0px; margin: 0px; z-index: 0; position: relative; color: var(--color-text-primary); display: flex; flex-direction: column; list-style-type: none;"><li title="" role="treeitem" class="parent object-properties-section-root-element" aria-expanded="false" style="box-sizing: border-box; min-width: 0px; min-height: 16px; display: flex; flex-direction: row; text-overflow: ellipsis; white-space: nowrap; position: relative; align-items: center; user-select: text;"><slot style="box-sizing: border-box; min-width: 0px; min-height: 0px;"><span class="console-object console-object-preview" tabindex="-1" style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: pre-wrap; overflow-wrap: break-word; font-style: italic; word-break: break-all;"><span class="object-properties-preview" style="box-sizing: border-box; min-width: 0px; min-height: 0px;">{<span class="name" style="box-sizing: border-box; min-width: 0px; min-height: 0px; color: var(--color-text-secondary); flex-shrink: 0;">query</span>: <span class="object-value-string" style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: pre-wrap; unicode-bidi: -webkit-isolate; color: var(--color-syntax-1); word-break: break-all;">'select * from c'</span>, <span class="name" style="box-sizing: border-box; min-width: 0px; min-height: 0px; color: var(--color-text-secondary); flex-shrink: 0;">pagingInfo</span>: <span class="object-value-object" title="Object" style="box-sizing: border-box; min-width: 0px; min-height: 0px;">{…}</span>}</span></span></slot></li></ol></div></div></span></span></div></div><div tabindex="-1" class="console-message-wrapper console-from-api console-info-level" style="box-sizing: border-box; min-width: 0px; min-height: 0px; display: flex; border-top: 1px solid var(--override-message-border-color); border-bottom: 1px solid transparent; --console-color-black: #000; --console-color-red: #a00; --console-color-green: #0a0; --console-color-yellow: #a50; --console-color-blue: #00a; --console-color-magenta: #a0a; --console-color-cyan: #0aa; --console-color-gray: #aaa; --console-color-darkgray: #555; --console-color-lightred: #f55; --console-color-lightgreen: #5f5; --console-color-lightyellow: #ff5; --console-color-lightblue: #55f; --console-color-ightmagenta: #f5f; --console-color-lightcyan: #5ff; --console-color-white: #fff; color: rgb(32, 33, 36); font-family: consolas, &quot;lucida console&quot;, &quot;courier new&quot;, monospace; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="console-message" style="box-sizing: border-box; min-width: 0px; min-height: 17px; clear: right; position: relative; padding: 3px 22px 1px 0px; margin-left: 24px; flex: 1 1 auto; display: flex;"><span class="source-code" style="box-sizing: border-box; min-width: 0px; min-height: 0px; font-family: var(--source-code-font-family); white-space: pre-wrap; font-size: var(--source-code-font-size) !important; flex: 1 1 auto; line-height: 1.2;"><span class="console-message-anchor" style="box-sizing: border-box; min-width: 0px; min-height: 0px; float: right; text-align: right; max-width: 100%; margin-left: 4px;"><span class="devtools-link" role="link" tabindex="-1" title="https://sqlopsbuilds.blob.core.windows.net/sourcemaps/4970733324ef8254b7c22a5dc55af7f8a1dea93f/core/vs/workbench/out-vscode/vs/workbench/file:/__w/1/s/src/vs/base/common/console.ts:137" style="box-sizing: border-box; min-width: 0px; min-height: 0px; color: var(--color-text-secondary); text-decoration: underline; outline-offset: 2px; cursor: pointer; word-break: break-all;">console.ts:137</span> </span><span class="console-message-text" style="box-sizing: border-box; min-width: 0px; min-height: 0px;"><span style="box-sizing: border-box; min-width: 0px; min-height: 0px; contain: paint; display: inline-block; max-width: 100%; color: blue;">[Extension Host] </span>onDidReceiveMessage <div class="console-view-object-properties-section object-value-object source-code" style="flex: 1 1 auto; padding: 0px; --monospace-font-size: 12px; --monospace-font-family: consolas,lucida console,courier new,monospace; --source-code-font-size: 12px; --source-code-font-family: consolas,lucida console,courier new,monospace; box-sizing: border-box; min-width: 0px; min-height: 0px; font-family: var(--source-code-font-family); white-space: pre-wrap; font-size: var(--source-code-font-size) !important; position: relative; vertical-align: baseline; color: inherit; display: inline-block; overflow-wrap: break-word; max-width: 100%; line-height: 1.2;"><div class="tree-outline-disclosure tree-outline-disclosure-hide-overflow" style="box-sizing: border-box; min-width: 0px; min-height: 0px;"><ol class="tree-outline hide-selection-when-blurred source-code object-properties-section" role="tree" tabindex="-1" style="box-sizing: border-box; min-width: 0px; min-height: 0px; font-family: var(--source-code-font-family); white-space: pre-wrap; font-size: var(--source-code-font-size) !important; padding: 0px; margin: 0px; z-index: 0; position: relative; color: var(--color-text-primary); display: flex; flex-direction: column; list-style-type: none;"><li title="" role="treeitem" class="parent object-properties-section-root-element" aria-expanded="false" style="box-sizing: border-box; min-width: 0px; min-height: 16px; display: flex; flex-direction: row; text-overflow: ellipsis; white-space: nowrap; position: relative; align-items: center; user-select: text;"><slot style="box-sizing: border-box; min-width: 0px; min-height: 0px;"><span class="console-object console-object-preview" tabindex="-1" style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: pre-wrap; overflow-wrap: break-word; font-style: italic; word-break: break-all;"><span class="object-properties-preview" style="box-sizing: border-box; min-width: 0px; min-height: 0px;">{<span class="name" style="box-sizing: border-box; min-width: 0px; min-height: 0px; color: var(--color-text-secondary); flex-shrink: 0;">action</span>: <span class="object-value-string" style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: pre-wrap; unicode-bidi: -webkit-isolate; color: var(--color-syntax-1); word-break: break-all;">'submitQuery'</span>, <span class="name" style="box-sizing: border-box; min-width: 0px; min-height: 0px; color: var(--color-text-secondary); flex-shrink: 0;">query</span>: <span class="object-value-object" title="Object" style="box-sizing: border-box; min-width: 0px; min-height: 0px;">{…}</span>}</span></span></slot></li></ol></div></div></span></span></div></div><div tabindex="-1" class="console-message-wrapper console-from-api console-info-level" style="box-sizing: border-box; min-width: 0px; min-height: 0px; display: flex; border-top: 1px solid var(--override-message-border-color); border-bottom: 1px solid transparent; --console-color-black: #000; --console-color-red: #a00; --console-color-green: #0a0; --console-color-yellow: #a50; --console-color-blue: #00a; --console-color-magenta: #a0a; --console-color-cyan: #0aa; --console-color-gray: #aaa; --console-color-darkgray: #555; --console-color-lightred: #f55; --console-color-lightgreen: #5f5; --console-color-lightyellow: #ff5; --console-color-lightblue: #55f; --console-color-ightmagenta: #f5f; --console-color-lightcyan: #5ff; --console-color-white: #fff; color: rgb(32, 33, 36); font-family: consolas, &quot;lucida console&quot;, &quot;courier new&quot;, monospace; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="console-message" style="box-sizing: border-box; min-width: 0px; min-height: 17px; clear: right; position: relative; padding: 3px 22px 1px 0px; margin-left: 24px; flex: 1 1 auto; display: flex;"><span class="source-code" style="box-sizing: border-box; min-width: 0px; min-height: 0px; font-family: var(--source-code-font-family); white-space: pre-wrap; font-size: var(--source-code-font-size) !important; flex: 1 1 auto; line-height: 1.2;"><span class="console-message-anchor" style="box-sizing: border-box; min-width: 0px; min-height: 0px; float: right; text-align: right; max-width: 100%; margin-left: 4px;"><span class="devtools-link" role="link" tabindex="-1" title="https://sqlopsbuilds.blob.core.windows.net/sourcemaps/4970733324ef8254b7c22a5dc55af7f8a1dea93f/core/vs/workbench/out-vscode/vs/workbench/file:/__w/1/s/src/vs/base/common/console.ts:137" style="box-sizing: border-box; min-width: 0px; min-height: 0px; color: var(--color-text-secondary); text-decoration: underline; outline-offset: 2px; cursor: pointer; word-break: break-all;">console.ts:137</span> </span><span class="console-message-text" style="box-sizing: border-box; min-width: 0px; min-height: 0px;"><span style="box-sizing: border-box; min-width: 0px; min-height: 0px; contain: paint; display: inline-block; max-width: 100%; color: blue;">[Extension Host] </span>submitquery <div class="console-view-object-properties-section object-value-object source-code" style="flex: 1 1 auto; padding: 0px; --monospace-font-size: 12px; --monospace-font-family: consolas,lucida console,courier new,monospace; --source-code-font-size: 12px; --source-code-font-family: consolas,lucida console,courier new,monospace; box-sizing: border-box; min-width: 0px; min-height: 0px; font-family: var(--source-code-font-family); white-space: pre-wrap; font-size: var(--source-code-font-size) !important; position: relative; vertical-align: baseline; color: inherit; display: inline-block; overflow-wrap: break-word; max-width: 100%; line-height: 1.2;"><div class="tree-outline-disclosure tree-outline-disclosure-hide-overflow" style="box-sizing: border-box; min-width: 0px; min-height: 0px;"><ol class="tree-outline hide-selection-when-blurred source-code object-properties-section" role="tree" tabindex="-1" style="box-sizing: border-box; min-width: 0px; min-height: 0px; font-family: var(--source-code-font-family); white-space: pre-wrap; font-size: var(--source-code-font-size) !important; padding: 0px; margin: 0px; z-index: 0; position: relative; color: var(--color-text-primary); display: flex; flex-direction: column; list-style-type: none;"><li title="" role="treeitem" class="parent object-properties-section-root-element" aria-expanded="false" style="box-sizing: border-box; min-width: 0px; min-height: 16px; display: flex; flex-direction: row; text-overflow: ellipsis; white-space: nowrap; position: relative; align-items: center; user-select: text;"><slot style="box-sizing: border-box; min-width: 0px; min-height: 0px;"><span class="console-object console-object-preview" tabindex="-1" style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: pre-wrap; overflow-wrap: break-word; font-style: italic; word-break: break-all;"><span class="object-properties-preview" style="box-sizing: border-box; min-width: 0px; min-height: 0px;">{<span class="name" style="box-sizing: border-box; min-width: 0px; min-height: 0px; color: var(--color-text-secondary); flex-shrink: 0;">query</span>: <span class="object-value-string" style="box-sizing: border-box; min-width: 0px; min-height: 0px; white-space: pre-wrap; unicode-bidi: -webkit-isolate; color: var(--color-syntax-1); word-break: break-all;">'select * from c'</span>, <span class="name" style="box-sizing: border-box; min-width: 0px; min-height: 0px; color: var(--color-text-secondary); flex-shrink: 0;">pagingInfo</span>: <span class="object-value-object" title="Object" style="box-sizing: border-box; min-width: 0px; min-height: 0px;">{…}</span>}</span></span></slot></li></ol></div></div></span></span></div></div><div tabindex="-1" class="console-message-wrapper console-from-api console-error-level console-selected" style="box-sizing: border-box; min-width: 0px; min-height: 0px; outline-width: 0px; display: flex; border-top-width: ; border-top-style: ; border-top-color: var(--override-error-border-color); border-bottom-width: 1px; border-bottom-style: solid; border-bottom-color: var(--override-error-border-color); --console-color-black: #000; --console-color-red: #a00; --console-color-green: #0a0; --console-color-yellow: #a50; --console-color-blue: #00a; --console-color-magenta: #a0a; --console-color-cyan: #0aa; --console-color-gray: #aaa; --console-color-darkgray: #555; --console-color-lightred: #f55; --console-color-lightgreen: #5f5; --console-color-lightyellow: #ff5; --console-color-lightblue: #55f; --console-color-ightmagenta: #f5f; --console-color-lightcyan: #5ff; --console-color-white: #fff; background-color: var(--override-focused-message-background-color); --override-error-text-color: rgb(200 0 0); color: rgb(32, 33, 36); font-family: consolas, &quot;lucida console&quot;, &quot;courier new&quot;, monospace; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="console-message" style="box-sizing: border-box; min-width: 0px; min-height: 17px; clear: right; position: relative; padding: 3px 22px 1px 0px; margin-left: 24px; flex: 1 1 auto; display: flex;"><devtools-icon class="message-level-icon" aria-label="Error" style="display: inline-block; white-space: nowrap; position: absolute; box-sizing: border-box; min-width: 0px; min-height: 0px; flex: 1 1 auto; left: -17px; top: 2px; user-select: none;"><div class="icon-basic" style="width: 14px; height: 14px; display: block; -webkit-mask-image: url(&quot;devtools://devtools/bundled/Images/cross-circle-filled.svg&quot;); -webkit-mask-position: center center; -webkit-mask-repeat: no-repeat; -webkit-mask-size: 99%; background-color: var(--icon-color, var(--icon-error));"></div></devtools-icon><div class="console-message-stack-trace-toggle" style="box-sizing: border-box; min-width: 0px; min-height: 0px; flex: 1 1 auto; display: flex; flex-direction: row; align-items: flex-start;"><div class="console-message-stack-trace-wrapper" aria-label="notificationsAlerts.ts:42 Cannot read properties of undefined (reading 'documents') Stack table expanded" style="box-sizing: border-box; min-width: 0px; min-height: 0px; flex: 1 1 auto; display: flex; flex-direction: column; align-items: stretch;"><div aria-expanded="true" tabindex="-1" style="box-sizing: border-box; min-width: 0px; min-height: 0px; flex: 0 0 auto;"><span is="ui-icon" class="console-message-expand-icon spritesheet-triangle-down triangle-down icon-mask" style="box-sizing: border-box; min-width: 0px; min-height: 0px; display: inline-block; flex-shrink: 0; margin-bottom: -4px; background-color: var(--icon-default); -webkit-mask-position: var(--spritesheet-position); -webkit-mask-image: var(--image-file-triangle-down); --spritesheet-position: 0px 14px; width: 14px; height: 14px;"></span><span class="source-code" style="box-sizing: border-box; min-width: 0px; min-height: 0px; font-family: var(--source-code-font-family); white-space: pre-wrap; font-size: var(--source-code-font-size) !important; line-height: 1.2;"><span class="console-message-anchor" style="box-sizing: border-box; min-width: 0px; min-height: 0px; float: right; text-align: right; max-width: 100%; margin-left: 4px;"><span class="devtools-link" role="link" tabindex="-1" title="https://sqlopsbuilds.blob.core.windows.net/sourcemaps/4970733324ef8254b7c22a5dc55af7f8a1dea93f/core/vs/workbench/out-vscode/vs/workbench/file:/__w/1/s/src/vs/workbench/browser/parts/notifications/notificationsAlerts.ts:42" style="box-sizing: border-box; min-width: 0px; min-height: 0px; color: var(--color-text-secondary); text-decoration: underline; outline-offset: 2px; cursor: pointer; word-break: break-all;">notificationsAlerts.ts:42</span> </span><span class="console-message-text" style="box-sizing: border-box; min-width: 0px; min-height: 0px; color: var(--override-error-text-color) !important;">Cannot read properties of undefined (reading 'documents')</span></span></div><div class="" role="group" style="box-sizing: border-box; min-width: 0px; min-height: 0px; flex: 0 0 auto;"><span class="monospace" style="display: inline-block; --monospace-font-size: 12px; --monospace-font-family: consolas,lucida console,courier new,monospace; --source-code-font-size: 12px; --source-code-font-family: consolas,lucida console,courier new,monospace; box-sizing: border-box; min-width: 0px; min-height: 0px; font-family: var(--monospace-font-family); font-size: var(--monospace-font-size) !important;">
  | onDidChangeNotification | @ | notificationsAlerts.ts:42
-- | -- | -- | --
  | (anonymous) | @ | notificationsAlerts.ts:28
  | _deliver | @ | event.ts:1109
  | _deliverQueue | @ | event.ts:1120
  | fire | @ | event.ts:1144
  | addNotification | @ | notifications.ts:204
  | notify | @ | notificationService.ts:175
  | (anonymous) | @ | mainThreadMessageService.ts:88
  | ZoneAwarePromise | @ | C:\Program Files\Azu…s\dist\zone.js:1351
  | _showMessage | @ | mainThreadMessageService.ts:44
  | $showMessage | @ | mainThreadMessageService.ts:38
  | _doInvokeHandler | @ | rpcProtocol.ts:457
  | _invokeHandler | @ | rpcProtocol.ts:442
  | _receiveRequest | @ | rpcProtocol.ts:370
  | _receiveOneMessage | @ | rpcProtocol.ts:296
  | (anonymous) | @ | rpcProtocol.ts:161
  | _deliver | @ | event.ts:1109
  | fire | @ | event.ts:1140
  | fire | @ | ipc.net.ts:650
  | J.onmessage | @ | localProcessExtensionHost.ts:586

</span></div></div></div></div></div>
</body>
</html>

**Comment by JoeFieck** [comment]
@languy Here is the console output in a screen shot if this helps interpret with the format at all...

![image](https://github.com/Azure/azure-cosmosdb-ads-extension/assets/148320389/caf712e3-4fbb-4007-a5fa-76b9b3e24d49)

**Comment by JoeFieck** [comment]
Here is the screen shot show that from Azure Portal we get the records...

![image](https://github.com/Azure/azure-cosmosdb-ads-extension/assets/148320389/2d320062-a308-4d8c-aea4-be670df50e82)

**Comment by JoeFieck** [comment]
Also @languy I'm not sure this helps but here is the notificationsAlert.ts typescript file throwing the error...

/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Microsoft Corporation. All rights reserved.
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

import { alert } from 'vs/base/browser/ui/aria/aria';
import { localize } from 'vs/nls';
import { INotificationViewItem, INotificationsModel, NotificationChangeType, INotificationChangeEvent, NotificationViewItemContentChangeKind } from 'vs/workbench/common/notifications';
import { Disposable } from 'vs/base/common/lifecycle';
import { toErrorMessage } from 'vs/base/common/errorMessage';
import { NotificationPriority, Severity } from 'vs/platform/notification/common/notification';
import { Event } from 'vs/base/common/event';

export class NotificationsAlerts extends Disposable {

	constructor(private readonly model: INotificationsModel) {
		super();

		// Alert initial notifications if any
		for (const notification of model.notifications) {
			this.triggerAriaAlert(notification);
		}

		this.registerListeners();
	}

	private registerListeners(): void {
		this._register(this.model.onDidChangeNotification(e => this.onDidChangeNotification(e)));
	}

	private onDidChangeNotification(e: INotificationChangeEvent): void {
		if (e.kind === NotificationChangeType.ADD) {

			// ARIA alert for screen readers
			this.triggerAriaAlert(e.item);

			// Always log errors to console with full details
			if (e.item.severity === Severity.Error) {
				if (e.item.message.original instanceof Error) {
					console.error(e.item.message.original);
				} else {
					console.error(toErrorMessage(e.item.message.linkedText.toString(), true));
				}
			}
		}
	}

	private triggerAriaAlert(notification: INotificationViewItem): void {
		if (notification.priority === NotificationPriority.SILENT) {
			return;
		}

		// Trigger the alert again whenever the message changes
		const listener = notification.onDidChangeContent(e => {
			if (e.kind === NotificationViewItemContentChangeKind.MESSAGE) {
				this.doTriggerAriaAlert(notification);
			}
		});

		Event.once(notification.onDidClose)(() => listener.dispose());

		this.doTriggerAriaAlert(notification);
	}

	private doTriggerAriaAlert(notification: INotificationViewItem): void {
		let alertText: string;
		if (notification.severity === Severity.Error) {
			alertText = localize('alertErrorMessage', "Error: {0}", notification.message.linkedText.toString());
		} else if (notification.severity === Severity.Warning) {
			alertText = localize('alertWarningMessage', "Warning: {0}", notification.message.linkedText.toString());
		} else {
			alertText = localize('alertInfoMessage', "Info: {0}", notification.message.linkedText.toString());
		}

		alert(alertText);
	}
}

**Comment by languy** [comment]
Thanks for the detailed information. Unfortunately, it looks like the stack trace shows the code that displays the error message in ADS (`NotificationsAlerts`), not the code that actually triggered the exception in the extension.

This said, your log provides a few clues: the last successful console log is "submitquery" and the error message shows that it's trying to read the `documents` property of an `undefined` object. [This](https://github.com/Azure/azure-cosmosdb-ads-extension/blob/4d379ddb101b81d48252655a709b47f5690f379f/src/extension.ts#L860) is one place where it could happen, but my guess is it's happening inside [this call](https://github.com/Azure/azure-cosmosdb-ads-extension/blob/4d379ddb101b81d48252655a709b47f5690f379f/src/extension.ts#L852) which also reads the `documents` property of the result. The result of a query should always be an object (maybe an empty array if no results), but never `undefined`. Why would the proxy return an `undefined` query result, then? The query is executed by a binary proxy implemented in C# which is hard to debug in your production build. One possible reason: you have multiple proxies running at the same time against the same database which yields `undefined` result. That seems consistent with what you reported: maybe the 2nd time you pressed, it spawned another proxy while the proxy from the first click was still executing.

Let's try this:
* Install the latest version 0.4.3 of the extension. It likely won't resolve the issue, but there are some improvements/bug fixes in the query editor UI and at least you'll be running the latest version.
* Reboot and restart ADS. It will ensure that you have only one instance of ADS and one instance of the proxy running on your machine.
* Click "Run" only once. There will be a progress indicator. Wait for it to complete. It has to complete either by showing the results or by failing with some error.

If that does not resolve the issue, we may have to understand why the query is failing or hanging. [This](https://github.com/Azure/azure-cosmosdb-ads-extension/blob/4d379ddb101b81d48252655a709b47f5690f379f/CosmosDbProxy/CosmosDbProxy/SdkRpcTarget.cs#L210) is where the query is executed using the C# SDK.
* Were you ever able to run the query on other databases in your Cosmos DB account or other Cosmos DB account databases?
* How long does the query take on the portal Data Explorer? (it might slower on your desktop?)
* Is there any special configuration that might affect accessing the data? If so, can you repro on a test account with the default setup?
* Try a query that doesn't access the data? Such as `SELECT count(1)`
* Can you confirm you are on Windows?

**Comment by JoeFieck** [comment]
Hello @languy .  Here are my responses to your requests and suggestions...

I was using the 0.4.3 extension already but I went ahead and uninstalled and installed it new again just for good measure.
I rebooted and restarted ADS.  I then opened a query window and ran the "select * from c" query just once...
I do not see any type of progress bar on it that ever needed to finish.  I let the first run sit for like 20 minutes but nothing ever returned.  Here is how the screen looked:

![image](https://github.com/Azure/azure-cosmosdb-ads-extension/assets/148320389/1f5d9f3d-1aad-4a3d-a480-b4e01ca015ed)

**Comment by JoeFieck** [comment]
To your suggestion about running "select count(1)" from ADS extension.  I did try that and the results are the same.
I can run it once and nothing happens then I run it again after some time and I get this message saying :

![image](https://github.com/Azure/azure-cosmosdb-ads-extension/assets/148320389/cbde4385-061a-4cde-825d-be3718c8a390)

**Comment by JoeFieck** [comment]
Were you ever able to run the query on other databases in your Cosmos DB account or other Cosmos DB account databases?
Me : No.  I tried on two seperate Cosmos servers with the same behavior.  Seems isolated to the extension I think.

How long does the query take on the portal Data Explorer? (it might slower on your desktop?)
Me: From Azure Data Explorer the query runs pretty fast.  I'd say maybe 2 seconds.

Is there any special configuration that might affect accessing the data? If so, can you repro on a test account with the default setup?
Me: I have been able to reproduce this on two separate instances of Cosmo from here.

Try a query that doesn't access the data? Such as SELECT count(1)
Me: I tried and got same results.  Nothing back then run again and get error.

Can you confirm you are on Windows?
Me. Yes I'm on Windows.


Just a comment here.   As these things go our back and forth exchanges here on github are heavily paced by our time zone differences.  We'd probably triple our productivity if we coordinated a time when we could do a web meeting and share my screen.  I may even then be able to try connecting to a test Cognos server there on your side remotely from here and see if we can trigger the error

**Comment by languy** [comment]
> I may even then be able to try connecting to a test Cognos server there on your side remotely from here and see if we can trigger the error

Yes, please go in the portal and open your cosmos db account. Under "Support + Troubleshooting" (the last tab on the left), please file a ticket, so we can contact you directly. Thanks.

**Comment by JoeFieck** [comment]
@languy It not letting me create the ticket. It just keeps taking me around in circles but never shows me any "create ticket" or anything like that.  I'm here...

![image](https://github.com/Azure/azure-cosmosdb-ads-extension/assets/148320389/0ea06bcd-196d-47dd-97f7-25fec3f74b0c)

**Comment by JoeFieck** [comment]
I am fairly sure this is checking for problems on the instance of Cosmos running on Azure not for any issues from a remote client.  I think that could be why its not finding anything wrong and offering me a button to create a ticket.  Maybe anyway...

**Comment by JoeFieck** [comment]
Actually @languy its telling me this...

![image](https://github.com/Azure/azure-cosmosdb-ads-extension/assets/148320389/2e3c0938-5976-40bd-a66c-d515373246f0)

We have another plan under Xen.ai account that is paid.  But I can't login to that one here.  Is this going to be possible to continue without an incident?

**Comment by JoeFieck** [comment]
Don't give up on me @languy :-)  .   If you google this error there are a ton of people in the user community seeing this issue.  None ever having gotten to any real fix.  Maybe you could create the ticket for me and we could further our communications here on Github as this seems to reach us both okay for now.  What do you think?

**Comment by languy** [comment]
@joefieck Can you please fill out [this form](https://forms.office.com/r/HQEBmpMrSh)? and we will contact you directly. Thanks.

**Comment by JoeFieck** [comment]
Hi @languy .  I have filled out the form.  This was how the confirmation screen looked after I hit Submit.  No case or anything...

![image](https://github.com/Azure/azure-cosmosdb-ads-extension/assets/148320389/980d16f7-b3c8-40f4-abe5-2095535962ab)

**Comment by jbriggs22** [comment]
I was experiencing this issue after switching to MacBook and I had given up based on seeing this thread. Then later this week I installed the dotnet Runtime and SDK (dotnet 8 and dotnet 6) and had added it to PATH. Then tried the cosmosDb extension again and now it has been working fine since.

Looking at the source code for this extension, I see it's using dotnet for the CosmosDbProxy. Is it possible that missing dotnet (or not finding it on the PATH) is the cause?

Also, to note that I did have the .NET 6 Runtime Azure Data Studio Extension installed prior to installing the SDKs. But I don't think I was able to run just the `dotnet` command in a terminal.

Hope this helps.

**Comment by JoeFieck** [comment]
Hi @jbriggs22 .  From my talks with @languy we think this is related to the dotnet sdk used by ADT.  What is odd is that ADS is able to find all the containers and databases then show them in the object browser.  Just that on my P.C. using ADS it can't run a query.   Even a query that doesn't try to access a container.  I would think for my client to know to show me all three containers in the object browser it would still need dotnet connector.  If so its very peculiar that it would work there then fail when running a query.  

From the thread here I did install the Azure Cosmos DB extension into VS Code and from there I can run queries.  VS Code uses the JavaScript SDK and that works fine.

**Comment by JoeFieck** [comment]
Well @jbriggs22 I think you were right.  I installed a tool called Cosmos DB Explorer on Windows and it made me install a new dotnet driver.  After doing so the my connection to Cosmos in ADP started working.  Was the dotnet driver as you said.  Closing thread now.  And thanks!

**Comment by JoeFieck** [comment]
Hello all.  We can shut down this case.  I installed Windows based Cosmos
DB Explorer and it made me install a new dotnet driver which magically got
the Cosmos extension in ADP working again.  Thanks for the help...

**Comment by fa-bwilliams** [comment]
I am experiencing this exact issue today. I am using MacOS. I have version.1.48.0 of ADS and v0.4.3 for Cosmos DB extension. I do not know why you would suggest shutting down the case. It works properly for me too if using the DB Explorer via Azure Portal. However, it is still broken in ADS.

**Comment by JoeFieck** [comment]
Hi @fa-bwilliams .  This thread got mixed up with a side one that I was addressing with @languy offline via e-mail.  By "shut down the case" I was saying we could discontinue the e-mail threads.  Not that we should forget about this.  It is a bug, they have identified the fix and to the best of my knowledge are working on it.  And just heads up, VS Codes Cosmos extension works.  I also installed one called "Cosmos DB Explorer" that has the features of Azures "Data Explorer".  Both worked if you need something to use until ADS extension fix is out.

**Comment by languy** [comment]
Hi all, there is a bug in the way the extension spawns the proxy (a separate executable that actually executes the query). It does not correctly use the .NET extension and fails to spawn the proxy. The workaround is to manually install the .NET runtime or sdk (or install an app that itself installs the runtime).
Sorry for the inconvenience.

**Comment by languy** [comment]
The fix for this issue has just been released in v0.4.4.
